template/server.js

```diff
@@ -1,106 +1,110 @@
-// both bundles are built with webpack
 let renderer;
-if (isProduction) {
-  renderer = createRenderer(fs.readFileSync(
-    resolve('./dist/server-bundle.js'), 
-    'utf-8'
-  ));
-} else {
-  require('./build/dev-server')(application, bundle => {
-    renderer = createRenderer(bundle);
+let readyPromise;
+const templatePath = resolve('./src/index.template.html');
+if (isProd) {
+  const template = fs.readFileSync(templatePath, 'utf-8');
+  const bundle = require('./dist/vue-ssr-server-bundle.json');
+  const clientManifest = require('./dist/vue-ssr-client-manifest.json');
+
+  renderer = createRenderer(bundle, {
+    template,
+    clientManifest,
   });
+} else {
+  readyPromise =  require('./build/setup-dev-server')(
+    app,
+    templatePath,
+    (bundle, options) => {
+      renderer = createRenderer(bundle, options);
+    }
+  )
 }
 
-application.use(favicon(resolve('./static/favicon.ico')));
-if(isProduction) {
-  application.use('/static', express.static(resolve('./dist/static')));
-}
+const serve = (path, cache) => express.static(resolve(path), {
+  maxAge: cache && isProd ? 1000 * 60 * 60 * 24 * 30 : 0,
+});
 
-// writes the head of index.html
-// then injects the initial state of the application
-// then writes the application as it's being rendered
-// finally writes the tail of index.html
-application.get('*', (request, response) => {
-  if (!renderer) {
-    return response.end('Compiling... Refresh in a moment!');
+app.use(compression({ threshold: 0  }));
+app.use(favicon(resolve('./public/favicon.ico')));
+app.use('/dist', serve('./dist', true));
+app.use('/public', serve('./public', true));
+app.use('/manifest.json', serve('./manifest.json', true));
+app.use('/service-worker.js', serve('./dist/service-worker.js'));
+
+app.use(microcache.cacheSeconds(1, req => useMicroCache && req.originalUrl));
+
+function render (req, res) {
+  const s = Date.now();
+
+  res.setHeader("Content-Type", "text/html");
+  res.setHeader("Server", serverInfo);
+
+  const handleError = err => {
+    if (err.url) {
+      res.redirect(err.url)
+    } else if (err.code === 404) {
+      res.status(404).send('404 | Page Not Found');
+    } else {
+      // Render Error Page or Redirect
+      res.status(500).send('500 | Internal Server Error');
+      console.error(`error during render : ${req.url}`);
+      console.error(err.stack);
+    }
   }
 
-  const timestamp = Date.now();
-  let context = { url: request.originalUrl };
-  const renderStream = renderer.renderToStream(context);
-  let firstChunk = true;
-
-  response.write(indexHTML.head);
-
-  renderStream.on('data', chunk => {
-    if(firstChunk) {
-      // embed initial store state
-      if(context.initialState) {
-        response.write(
-          `<script>
-             window.INITIAL_STATE = ${serialize(context.initialState, { isJSON: true })};
-           </script>`
-        );
-      }
-
-      firstChunk = false;
+  const context = {
+    title: 'Big Spaceship Vue Template',
```

Comment:
Can/should the title be set when you are spinning up a new project (like how Yeoman does it) or in a more central config file?